lib/game_core.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module IcodebreakerGem
+  class GameCore
+    include Validation
+
+    attr_reader :secret
+
+    def initialize(secret)
+      validate_code secret
+
+      @secret = secret.to_s
+      @shuffle = @secret.each_char.to_a.shuffle
+      @hint_idx = 0
+    end
+
+    def compare(guess)
+      return '++++' if @secret == guess
+      return '' if (@secret.each_char.to_a & guess.each_char.to_a).empty?
```

Comment:
1) please use `secret` instead of `@secret` as you defined `attr_reader` for it
2) `secret` can't be empty by default.
3)  `return '' if guess.blank?`. Read the diff between `empty?` and `blank?`